conbench/entities/run.py

```diff
@@ -436,50 +356,59 @@ def _guts(cinfo: TypeCommitInfoGitHub) -> Tuple[Commit, bool]:
 
 
 class _Serializer(EntitySerializer):
-    def _dump(self, run: Run, get_baseline_runs: bool = False):
-        if run.commit:
-            commit = CommitSerializer().one.dump(run.commit)
-            commit.pop("links", None)
+    def _dump(
+        self, benchmark_result: "BenchmarkResult", get_baseline_runs: bool = False
+    ):
+        if benchmark_result.commit:
+            commit_dict = CommitSerializer().one.dump(benchmark_result.commit)
+            commit_dict.pop("links", None)
             commit_link = f.url_for(
-                "api.commit", commit_id=commit["id"], _external=True
+                "api.commit", commit_id=commit_dict["id"], _external=True
             )
         else:
-            commit = None
+            commit_dict = None
             commit_link = None
 
-        hardware = HardwareSerializer().one.dump(run.hardware)
-        hardware.pop("links", None)
-        result = {
-            "id": run.id,
-            "name": run.name,
-            "reason": run.reason,
+        hardware_dict = HardwareSerializer().one.dump(benchmark_result.hardware)
+        hardware_dict.pop("links", None)
+        out_dict = {
+            "id": benchmark_result.run_id,
+            "tags": benchmark_result.run_tags,
+            "reason": benchmark_result.run_reason,
+            "name": benchmark_result.run_tags.get("name"),
             "timestamp": conbench.util.tznaive_dt_to_aware_iso8601_for_api(
-                run.timestamp
-            ),
+                benchmark_result.run_tags["timestamp"]
+            )
+            if "timestamp" in benchmark_result.run_tags
+            else None,
             "finished_timestamp": conbench.util.tznaive_dt_to_aware_iso8601_for_api(
-                run.finished_timestamp
+                benchmark_result.run_tags["finished_timestamp"]
             )
-            if run.finished_timestamp
+            if "finished_timestamp" in benchmark_result.run_tags
             else None,
-            "info": run.info,
-            "error_info": run.error_info,
-            "error_type": run.error_type,
-            "commit": commit,
-            "hardware": hardware,
-            "has_errors": run.has_errors,
+            "info": benchmark_result.run_tags.get("info"),
+            "error_info": benchmark_result.run_tags.get("error_info"),
+            "error_type": benchmark_result.run_tags.get("error_type"),
+            "commit": commit_dict,
+            "hardware": hardware_dict,
+            "has_errors": benchmark_result.run_tags.get("has_errors"),
```

Comment:
This will also be interesting.

Current behavior: a run has `has_errors` set to true when any benchmark result in the run is failed. We do this here:

https://github.com/conbench/conbench/blob/42ef379825e0bb8d7cedc00f0d143e84f102c60b/conbench/entities/benchmark_result.py#L237

For maintaining that behavior we cannot look at just one result.

Reply:
In this PR I'm basically dropping all first-class support for these old Run fields:
- finished_timestamp
- error_info
- error_type
- has_errors

So an "errored run" is not an official thing anymore. (People should still be able to hit `/api/benchmark-results/?run_id=ABC` and get `[]`, or `/api/runs/ABC` and get a 404, which may indicate something bad happened if they expected to find benchmark results.)

If people really want to submit those fields under `run_tags` with a BenchmarkResult, they can, but to me that doesn't make sense anymore. I think I'm actually going to drop those from this API run JSON instead of looking in `run_tags` for them.

(I'm also dropping support for `Run.timestamp`, which was never mutable but instead an insertion time. I'll keep that in the API run JSON for legacy reasons, by using the earliest benchmark result's timestamp.)

Reply:
Okay, update: with the new way we query for runs from the database when populating the Runs UI page, it's basically trivial to aggregate them. So I decided to keep the notion of an "errored run," except now it's scoped to
- viewing the Runs UI page (nowhere else)
- being True if and only if there's at least one associated benchmark result that `is_failed`